lib/metasploit/framework/login_scanner/http.rb

```diff
@@ -441,10 +452,22 @@ def set_sane_defaults
 
         # Combine the base URI with the target URI in a sane fashion
         #
-        # @param [String] target_uri the target URL
+        # @param [Array<String>] target_uri the target URL
         # @return [String] the final URL mapped against the base
-        def normalize_uri(target_uri)
-          (self.uri.to_s + "/" + target_uri.to_s).gsub(/\/+/, '/')
+        def normalize_uri(*target_uri)
+          if target_uri.count == 1
+            (uri.to_s + '/' + target_uri.first.to_s).gsub(%r{/+}, '/')
+          else
+            new_str = target_uri * '/'
+            new_str = new_str.gsub!('//', '/') while new_str.index('//')
+
+            # Makes sure there's a starting slash
+            unless new_str[0,1] == '/'
+              new_str = '/' + new_str
+            end
+
+            new_str
+          end
```

Comment:
what's the reason for this change? `uri` needs to be set correctly anyway right since we use it throughout the login scanner?

Reply:
yea I did spot that this was the only one implemented this way which I thought was weird, I do kind of understand it though, since it should always be building on top of `uri` 

do you remember any specific examples of things breaking that you ran into? I wonder if theres another issue since I don't think I've ran into that particular one before

Reply:
The behavior is like this:

- When invoked from framework, the module includes the `rest_client` and `jenkins` mixin, making the original path work
- When invoked from Pro however the `rest_client` mixin is not included and causes the code to fail when trying to call anything inside the `jenkins` mixin.

This worked originally because the framwork scanner hacked around this by passing a block, and not calling the internal methods such as `send_request_cgi`, because those methods are not defined.
The including module in framework includes this because the mixin was defined for that behavior, but not for the actual scanner itself:

- no `send_request_cgi` method
- no DATASTORE